### ISSUE TITLE
Blockstorage V2: panic: interface conversion : interface {} is nil, not []interface {}

### ISSUE THREAD

**Issue by waynr** [gophercloud/gophercloud#1269]
Before starting a PR, please read our [contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

```
panic: interface conversion: interface {} is nil, not []interface {}                     
                                                                                         
goroutine 1 [running]:                                                                                                                                                                                                                                                        
github.com/rcbops/mk8s/vendor/github.com/gophercloud/gophercloud.Result.extractIntoPtr(0xc000317b30, 0xc00056c2a0, 0xc00056c360, 0x0, 0x0, 0x103aba0, 0xc0003ce320, 0x1233e24, 0x7, 0x0, ...)                                                                                    
        /go/src/github.com/rcbops/mk8s/vendor/github.com/gophercloud/gophercloud/results.go:93 +0xc18                                                                                                                                                                            
github.com/rcbops/mk8s/vendor/github.com/gophercloud/gophercloud.Result.ExtractIntoSlicePtr(0xc000317b30, 0xc00056c2a0, 0xc00056c360, 0x0, 0x0, 0x103aba0, 0xc0003ce320, 0x1233e24, 0x7, 0xc00014b418, ...)
        /go/src/github.com/rcbops/mk8s/vendor/github.com/gophercloud/gophercloud/results.go:200 +0x1ba                                                                                
github.com/rcbops/mk8s/vendor/github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes.ExtractVolumesInto(0x1390260, 0xc0005ac300, 0x103aba0, 0xc0003ce320, 0x1390260, 0xc0005ac0c0)                                                                                 
        /go/src/github.com/rcbops/mk8s/vendor/github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes/results.go:146 +0xd6
```

I have tracked the issue down to this exact commit: https://github.com/gophercloud/gophercloud/commit/f1bae72a73954e0636d2de309b417bcfa4bd7de7

I have confirmed that this commit is what leads to the panic shown above in two ways. First off, the commit immediately preceeding it does not exhibit the behavior. Second, I have reverted it in the latest master branch and the problem goes away.

Here is the code block which results in the above stack trace:

```
type VolumeWithTenant struct {
	volumes.Volume
	volumetenants.VolumeTenantExt
}

func getVolumes(client *gophercloud.ServiceClient) ([]VolumeWithTenant, error) {
	allPages, err := volumes.List(client, volumes.ListOpts{}).AllPages()
	if err != nil {
		return nil, err
	}

	var allVolumes []VolumeWithTenant

	err = volumes.ExtractVolumesInto(allPages, &allVolumes)
	if err != nil {
		return nil, err
	}

	return allVolumes, nil
}
```

The problem seems to be that `allPages` here is empty? At least, I don't see any volumes in use in the tenant I'm currently using. When I `fmt.Println(allPages` this is what I get: ```{{{{map[:[]] map[] <nil>} {      false  }} []}}```

I guess I should be calling the `IsEmpty` method on `allPages` before attempting to extract it and return `nil` in the first return value position if it is. But I'm still filing this ticket just in case someone else comes across this behavior.

**Comment by jrperritt** [comment]
Hmm, this is something the library should handle for you (but is not currently implemented). I think adding some logic here: https://github.com/gophercloud/gophercloud/blob/master/results.go#L199 to return if length zero would do it.

**Comment by rickard-von-essen** [comment]
@jrperritt hashicorp/packer#6885 seems similar, should I open another issue for that?

**Comment by jrperritt** [comment]
closed by #1305
